docs/review.md

```diff
@@ -0,0 +1,124 @@
+  for an easy shortcut).
+- Compile the code. See [INSTALL.md](INSTALL.md) and [MAKEFILE.md](MAKEFILE.md)
+  for inputs on how to do that.
+- Run the unit tests of the changed areas. See the [`unit` section of
+  MAKEFILE.md](MAKEFILE.md#unit) on how to run specific packages or test cases
+  directly.
+- Run the code on your local machine and attempt to test the changes of the
+  pull request. For any actual Bitcoin or Lightning related functionality, you
+  might want to spin up a `regtest` local network to test against. You can check
+  out the [`regtest` setup project](https://github.com/guggero/rt) that might
+  be helpful for that.
+- Try to manually test the changes in the PR.
+- Keep the logs of your tests as part of your review feedback. If you find a bug
+  the exact steps to reproduce as well as the local logs will be super helpful
+  to the author of the Pull Request.
+- Propose areas where the unit test coverage could be improved (try to propose
+  specific test cases, for example "we should test the behavior with a negative
+  number here", instead of just saying "this area needs more tests").
+- Bonus: Write a unit test (or test case) that you can propose to add to the PR.
+- Go through the PR commit by commit and line by line
+    - Is the flow of the commits easy to understand? Does the order make sense?
+      See the [code contribution guidelines](code_contribution_guidelines.md)
+      for more information on this.
+    - Is the granularity of the commit structure easy to understand? Should
+      commits be split up or be combined?
+    - What does each line achieve in terms of the overall goal of the PR?
+    - What does each line change in the behavior of the code?
+    - Are there any potentially unwanted side effects?
+    - Are all errors handled appropriately?
+    - Is there sufficient logging to trace what happens if a user runs into
+      problems with the code? (e.g. when testing the code, can you find out what
+      code path your execution took just by looking at the trace-level log?)
+    - Does the commit message appropriately reflect the changes in that commit?
+      Only relevant for non-self-explanatory commits
+- Is the documentation added by the PR sufficient?
+- Bonus: Propose additional documentation as part of the review that can be
+  added to the PR
+- What does the user interface (command line flags, gRPC/REST API) affected by
+  the PR look like? Is it easy to understand for users?
+- Does the CI pass for the PR? If not, can you find out what needs to be changed
+  to make the CI pass? Unfortunately there are still some flakes in the tests,
+  so some steps might fail unrelated to the changes in the Pull Request.
+
+## How to submit review feedback
+
+The above checklist is very detailed and extensive. But a review can also be
+helpful if it only covers some of the steps outlined. The more, the better, of
+course, but any review is welcome.
```

Comment:
wondering if we should add something that hints at the fact that review that is only making nit/style suggestions is maybe not that useful? just trying to prevent a bunch of review noise

Reply:
Hmm, yeah, good point. Even though IMO some spam in PRs is easier to ignore than spam PRs. Added a sentence to that effect anyway.